projects/igniteui-angular/src/lib/grids/grid-public-row.ts

```diff
@@ -0,0 +1,492 @@
+import { IgxRow } from './common/crud.service';
+import { RowPinningPosition } from './common/enums';
+import { GridType } from './common/grid.interface';
+import { RowType } from './common/row.interface';
+import { IgxGridBaseDirective } from './grid-base.directive';
+import { IgxGridAPIService } from './grid/grid-api.service';
+import { IgxGridComponent } from './grid/grid.component';
+import { IgxHierarchicalGridComponent } from './hierarchical-grid/hierarchical-grid.component';
+import { IgxSummaryResult } from './summaries/grid-summary';
+import { IgxTreeGridComponent } from './tree-grid/tree-grid.component';
+import { ITreeGridRecord } from './tree-grid/tree-grid.interfaces';
+
+export class IgxGridRow implements RowType {
+    /**
+     * Returns the view index calculated per the grid page.
+     */
+    public get viewIndex(): number {
+        return this.index + this.grid.page * this.grid.perPage;
+    }
+
+    /**
+     *  The data passed to the row component.
+     *
+     * ```typescript
+     * let selectedRowData = this.grid.selectedRows[0].rowData;
+     * ```
+     */
+    public get data(): any {
+        const grid = this.grid;
+        let allRowsData = grid.allRowsData;
+        if (!this._data) {
+            if (grid.pinnedRecordsCount) {
+                allRowsData = grid.pinning.rows !== RowPinningPosition.Bottom ?
+                    [...grid.pinnedRecords, ...grid.allRowsData] : [...grid.allRowsData, ...grid.pinnedRecords];
+            }
+        }
+        return this._data ?? allRowsData[this.index];
+    }
+
+    /**
+     * Needs to be removed !
+     *
+     * @hidden
+     */
+    public get rowData(): any {
+        return this.data;
+    }
+
+    /**
+     * Sets/gets whether the row is pinned.
+     * Default value is `false`.
+     * ```typescript
+     * this.grid.selectedRows[0].pinned = true;
+     * ```
+     */
+    public get pinned(): boolean {
+        return this.grid.isRecordPinned(this.data);
+    }
+
+    public set pinned(val: boolean) {
+        if (val) {
+            this.pin();
+        } else {
+            this.unpin();
+        }
+    }
+
+    /**
+     * Sets/gets whether the row is selected.
+     * Default value is `false`.
+     * ```typescript
+     * row.selected = true;
+     * ```
+     */
+    public get selected(): boolean {
+        return this.grid.selectionService.isRowSelected(this.rowID);
+    }
+
+    public set selected(val: boolean) {
+        if (val) {
+            this.grid.selectionService.selectRowsWithNoEvent([this.rowID]);
+        } else {
+            this.grid.selectionService.deselectRowsWithNoEvent([this.rowID]);
+        }
+        this.grid.cdr.markForCheck();
+    }
+
+    /**
+     * Sets whether this specific row has disabled functionality for editing and row selection.
+     * Default value is `false`.
+     * ```typescript
+     * const isDisabled = row.disabled;
+     * ```
+     */

```

Comment:
I guess the disabled state is active when a row is pinned and users have the 'ghost' instance from the unpinned area, right?
If that is the case, then this doc string should be more specific when and why a row is in a disabled state.